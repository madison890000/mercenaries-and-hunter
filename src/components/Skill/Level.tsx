import React from 'react';
import styles from './Level.module.scss';
import classNames from 'classnames';

const LevelUnderstand = () =>
    <div className={classNames(styles.level)}>
        <span />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <div className={styles.levelName}>Understand</div>
    </div>
const LevelFamiliar = () =>
    <div className={classNames(styles.level)}>
        <span />
        <span />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <div className={styles.levelName}>Familiar</div>
    </div>
const LevelProficient = () =>
    <div className={classNames(styles.level)}>
        <span />
        <span />
        <span />
        <span />
        <span className={styles.empty} />
        <div className={styles.levelName}>Proficient</div>
    </div>

const Level = () => null

Level.Understand = LevelUnderstand;
Level.Familiar = LevelFamiliar;
Level.Proficient = LevelProficient;

export default Level
