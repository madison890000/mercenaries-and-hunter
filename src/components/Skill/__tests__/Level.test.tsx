import React from 'react';
import Level from '../Level';
import renderer from 'react-test-renderer';

describe('renders Level ', () => {
    test('Level --->level = Understand', () => {
        const levelUnderstand = renderer.create(<Level.Understand />).toJSON();
        expect(levelUnderstand).toMatchSnapshot();
    });

    test('Level --->level = Familiar', () => {
        const levelUnderstand = renderer.create(<Level.Familiar />).toJSON();
        expect(levelUnderstand).toMatchSnapshot();
    });

    test('Level --->level = Proficient', () => {
        const levelUnderstand = renderer.create(<Level.Proficient />).toJSON();
        expect(levelUnderstand).toMatchSnapshot();
    });
});
