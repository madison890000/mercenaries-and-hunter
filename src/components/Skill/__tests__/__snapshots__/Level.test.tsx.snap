// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Level  Level --->level = Familiar 1`] = `
<div
  className="level"
>
  <span />
  <span />
  <span
    className="empty"
  />
  <span
    className="empty"
  />
  <span
    className="empty"
  />
  <div
    className="levelName"
  >
    Familiar
  </div>
</div>
`;

exports[`renders Level  Level --->level = Proficient 1`] = `
<div
  className="level"
>
  <span />
  <span />
  <span />
  <span />
  <span
    className="empty"
  />
  <div
    className="levelName"
  >
    Proficient
  </div>
</div>
`;

exports[`renders Level  Level --->level = Understand 1`] = `
<div
  className="level"
>
  <span />
  <span
    className="empty"
  />
  <span
    className="empty"
  />
  <span
    className="empty"
  />
  <span
    className="empty"
  />
  <div
    className="levelName"
  >
    Understand
  </div>
</div>
`;
